### PR TITLE
Feature/initial

### PR DESCRIPTION
**Pull request by st3v3nhunt**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    "nock": "^9.0.14",
+    "snyk": "^1.40.2"
+  },
+  "dependencies": {
+    "azure-storage": "^2.4.0",
+    "moment": "^2.18.1",
+    "nhsuk-bunyan-logger": "^1.4.0",
+    "node-schedule": "^1.2.4",
+    "pretty-hrtime": "^1.0.3",
+    "request-promise-native": "^1.0.5",
```
Comment (markysoft) [inline package.json]: there's a warning on install of warning "request-promise-native@1.0.5" has unmet peer dependency "request@^2.34". I think you also need to explicitly add request too.

File: config/config.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+function getMergedDataFilename() {
+  const prefix = process.env.NODE_ENV === 'production' ? '' : 'dev-';
```
Comment (markysoft) [inline config/config.js]: The environment check was removed in the sexual-health-information-etl by allowing the filename to be specified as an env var, and having it set with a `dev-` prefix in the local compose file. Might be worth doing the same here.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,57 @@
...
+Azure Storage account specified in `AZURE_STORAGE_CONNECTION_STRING`. The files
+will be uploaded into a container as specified in `AZURE_BLOB_CONTAINER_NAME`
+or `etl-ouput` as the default.  If the storage account is `primarycare` and the
+defaults were used the merged data set will be available to download from
+[https://primarycare.blob.core.windows.net/etl-output/merged-data.json](https://primarycare.blob.core.windows.net/etl-output/merged-data.json).
+
+## Running the application and scheduling
+
+The application will run at startup and then on a daily basis, while the
+container continues to run. . The time of day defaults to 7:15am, and can be
```
Comment (markysoft) [inline README.md]: Extra full stop

File: app.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+async function combine() {
+  if (!scheduleConfig.schedulerDisabled()) {
+    // run on initial start, then on the schedule
+    await mergeAndUpload();
+  }
+
+  log.info(`Scheduling sexual health service data merge with rule '${scheduleConfig.getSchedule()}'`);
+  schedule.scheduleJob(scheduleConfig.getSchedule(), async () => {
+    await mergeAndUpload();
```
Comment (markysoft) [inline app.js]: Exceptions thrown from the `await` are not handled and will result in an unhandled promise rejection, terminating the app.

File: config/config.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+module.exports = {
+  app: {
+    name: 'sexual-health-service-data-combiner',
+  },
+  blobContainerName: process.env.AZURE_BLOB_CONTAINER_NAME || 'etl-output',
+  data: {
+    dir: {
+      current: './data/current',
+      latest: './data/latest',
```
Comment (markysoft) [inline config/config.js]: The `current` and `latest` folders do not exist causing an exception to be thrown during download.

File: lib/utils/saveDataAndUploadToAzure.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+const azureService = require('./azureService');
```
Comment (markysoft) [inline lib/utils/saveDataAndUploadToAzure.js]: Once the `azure-data-service` package is available this and the the `azureService` could be replaced with the AzureDataService class

File: package.json (hunk 1 above)
Comment (st3v3nhunt) [reply package.json]: Yeah I think that is more a problem with the configuration of the package in question. https://github.com/request/request-promise-native/blob/master/package.json#L41 specifies `request@2.34` is required but I'd hazard a guess that it should be more lenient than that and probably be `^2.34.0` or something similar to signify any version 2.
More background on [peer dependencies](https://nodejs.org/en/blog/npm/peer-dependencies/).
Given request is installed, the package works and there are plenty of [issues](https://github.com/yarnpkg/yarn/issues/5347#issuecomment-376219812) about yarn not installing peer dependencies I'd be inclined to not do anything. Unless that isn't acceptable?

File: config/config.js (hunk 5 above)
Comment (st3v3nhunt) [reply config/config.js]: Mmh, I thought I'd tested that. I'll get it sorted.

File: lib/utils/saveDataAndUploadToAzure.js (hunk 6 above)
Comment (st3v3nhunt) [reply lib/utils/saveDataAndUploadToAzure.js]: Yeah, I was thinking the same.

**Pull request merged**